src/tabs/tab.directive.ts

```diff
@@ -12,13 +12,22 @@ import {
 } from '@angular/core';
 import { TabsetComponent } from './tabset.component';
 
-@Directive({ selector: 'tab, [tab]' })
+@Directive({
+  selector: 'tab, [tab]',
+  host: {
+    role: 'tabpanel'
+  }
+})
 export class TabDirective implements OnInit, OnDestroy {
   /** tab header text */
   @Input() heading: string;
-  /** tab id. The same id with suffix '-link' will be added to the corresponding &lt;li&gt; element  */
+  /** attribute id.  */
   @HostBinding('attr.id')
   @Input() id: string;
+  /** tab id. */
+  @HostBinding('attr.aria-labelledby')
+  @Input() tabId = `tab${Date.now()}`;
```

Comment:
remove this input and use the `id` we have right now (and set a default value for it)

Reply:
also replace `date.now()` with external variable that increments itself